lib/msf/core/auxiliary/report.rb

```diff
@@ -401,19 +402,25 @@ def store_loot(ltype, ctype, host, data, filename=nil, info=nil, service=nil)
     host = Msf::Util::Host.normalize_host(host)
 
     ws = (db ? myworkspace.name[0,16] : 'default')
-    name =
+    name = 
       Time.now.strftime("%Y%m%d%H%M%S") + "_" + ws + "_" +
       (host || 'unknown') + '_' + ltype[0,16] + '_' +
       Rex::Text.rand_text_numeric(6) + '.' + ext
-
     name.gsub!(/[^a-z0-9\.\_]+/i, '')
 
-    path = File.join(Msf::Config.loot_directory, name)
+    if changepath
+      loot_dir = Msf::Config.loot_directory + ::File::Separator + host
```

Comment:
To keep consistent, create a directory for every session that has downloaded files regardless of `changepath` is.

Check `changepath` variable and modify the filename according to the path or not, otherwise it looks very ugly and obscure.